fission/src/ui/modals/mirabuf/ImportMirabufModal.tsx

```diff
@@ -3,80 +3,142 @@ import Modal, { ModalPropsImpl } from "@/components/Modal"
+            setManifest(data)
+        })
+    }, [])
 
-                    setFolderData(await getFolderData(selectedProject, selectedFolder))
-                } else {
-                    console.log("Defaulting to project root folder")
-                    const data = await getFolderData(selectedProject, selectedProject.folder)
-                    console.log(`Folder Data:\n${JSON.stringify(data)}`)
-                    setFolderData(data)
+    let cachedRobotElements;
+    if (cachedRobots.length > 0) {
+        cachedRobotElements = cachedRobots.map(info =>
+            ItemCard({
+                name: info.name || info.cacheKey || "Unnamed Robot",
+                id: info.id,
+                buttonText: "import",
+                onClick: () => console.log(`Selecting cached robot: ${info.cacheKey}`),
+                secondaryButtonText: "delete",
+                secondaryOnClick: () => {
+                    console.log(`Deleting cache of: ${info.cacheKey}`)
+                    MirabufCachingService.Remove(info.cacheKey, info.id, MiraType.ROBOT)
+                }
+            })
+        )
+    }
+    let cachedFieldElements;
+    if (cachedFields.length > 0) {
+        cachedFieldElements = cachedFields.map(info =>
+            ItemCard({
+                name: info.name || info.cacheKey || "Unnamed Field",
+                id: info.id,
+                buttonText: "import",
+                onClick: () => console.log(`Selecting cached field: ${info.cacheKey}`),
+                secondaryButtonText: "delete",
+                secondaryOnClick: () => {
+                    console.log(`Deleting cache of: ${info.cacheKey}`)
+                    MirabufCachingService.Remove(info.cacheKey, info.id, MiraType.FIELD)
                 }
-            }
-        })()
-    }, [selectedProject, selectedFolder])
+            })
+        );
+    }
 
-    useEffect(() => {
-        if (folderData) {
-            console.log(`${folderData.length} items in folder data`)
-        } else {
-            console.log("No folder data")
-        }
-    }, [folderData])
+    let remoteRobots = manifest?.robots || [];
+    let remoteFields = manifest?.fields || [];
+    if (cachedRobots.length > 0) {
+        remoteRobots = remoteRobots.filter(path => !cachedRobots.some(info => info.cacheKey.includes(path)))
+    }
+    if (cachedFields.length > 0) {
+        remoteFields = remoteFields.filter(path => !cachedFields.some(info => info.cacheKey.includes(path)))
+    }
+    const remoteRobotElements = remoteRobots.map(path =>
+        ItemCard({
+            name: path,
+            id: path,
+            buttonText: "download",
+            onClick: () => console.log(`Selecting remote: ${path}`)
+        })
+    );
+
+    const remoteFieldElements = remoteFields.map(path =>
+        ItemCard({
+            name: path,
+            id: path,
+            buttonText: "download",
+            onClick: () => console.log(`Selecting remote: ${path}`)
```

Comment:
Rather than printing, if you replace these with the workings of the spawn modal, we can integrate the two. The idea was that anything that isn't cached would be cached and then, the next time you go to spawn modal, it will then be available as a cached item instead. This would also include looking out for duplicates (don't show robots available remotely if they are already in the cache), plus ensure that they also have the delete button.